extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -59,21 +62,56 @@ final class S3SeekableChannelProvider implements SeekableChannelsProvider {
     private final S3AsyncClient s3AsyncClient;
     private final S3Instructions s3Instructions;
 
-    private final Map<URI, Long> uriToFileSize;
+    private SoftReference<KeyedObjectHashMap<URI, FileSizeInfo>> uriToFileSizeSoftRef;
+
+    private static final class FileSizeInfo {
+        private final URI uri;
+        private final long size;
+
+        FileSizeInfo(@NotNull final URI uri, final long size) {
+            this.uri = Require.neqNull(uri, "uri");
+            this.size = size;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(uri, size);
+        }
+
+        @Override
+        public boolean equals(final Object other) {
+            if (this == other) {
+                return true;
+            }
+            if (other == null || getClass() != other.getClass()) {
+                return false;
+            }
+            final FileSizeInfo that = (FileSizeInfo) other;
+            return size == that.size && uri.equals(that.uri);
+        }
```

Comment:
Delete, unused, potentially create confusion.
```suggestion

        @Override
        public int hashCode() {
            return Objects.hash(uri, size);
        }

        @Override
        public boolean equals(final Object other) {
            if (this == other) {
                return true;
            }
            if (other == null || getClass() != other.getClass()) {
                return false;
            }
            final FileSizeInfo that = (FileSizeInfo) other;
            return size == that.size && uri.equals(that.uri);
        }
```